splunk-otel-android/src/main/java/com/splunk/rum/CarrierFinder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import android.os.Build;
+import android.telephony.TelephonyManager;
+import androidx.annotation.RequiresApi;
+
+@RequiresApi(api = Build.VERSION_CODES.P)
+class CarrierFinder {
+
+    private final TelephonyManager telephonyManager;
+
+    CarrierFinder(TelephonyManager telephonyManager) {
+        this.telephonyManager = telephonyManager;
+    }
+
+    Carrier get() {
+        Carrier.Builder builder = Carrier.builder();
+        int id = telephonyManager.getSimCarrierId();
+        builder.id(id);
+        CharSequence name = telephonyManager.getSimCarrierIdName();
+        if (validString(name.toString())) {
```

Comment:
Should probably be:
```suggestion
        if (validString(name)) {
```
I think you're getting an NPE in tests here